daemon/mgr/image.go

```diff
@@ -324,9 +328,48 @@ func (mgr *ImageManager) ListImages(ctx context.Context, filter filters.Args) ([
 }
 
 // SearchImages searches imaged from specified registry.
-func (mgr *ImageManager) SearchImages(ctx context.Context, name string, registry string) ([]types.SearchResultItem, error) {
+func (mgr *ImageManager) SearchImages(ctx context.Context, name, registry string, auth *types.AuthConfig) ([]types.SearchResultItem, error) {
 	// Directly send API calls towards specified registry
-	return nil, errtypes.ErrNotImplemented
+	if len(registry) == 0 {
+		registry = registry2.DefaultRegistry
+	}
+
+	u := registry + "search?q=" + url.QueryEscape(name)
+	req, err := http.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if auth != nil && auth.IdentityToken != "" && auth.Username != "" {
+		req.SetBasicAuth(auth.Username, auth.Password)
+	}
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("Unexepected status code %d", res.StatusCode)
```

Comment:
better not be capitalized when return err, since this err maybe wrapped into another err.

Reply:
please see https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
yeah